project/src/test/resources/docker/sql/create-schema.sql

```diff
@@ -469,3 +475,223 @@ CREATE TABLE dbo.policies(
+  courseAcr VARCHAR(50),
+  calendarSection VARCHAR(50),
+  calTerm VARCHAR(50),
+  lang VARCHAR(10),
+	categ VARCHAR(100))
+AS $$
+#print_strict_params ON
+DECLARE
+	clid INT;
+	cid INT;
+	cscal INT;
+  langid INT;
+  categoryid INT;
+BEGIN
+  IF (courseName IS NULL AND courseAcr IS NULL) THEN
+    RAISE 'For the Course parameters, you must provide at least a "name" or an "acronym"';
+  END IF;
+
+  SELECT id INTO langid FROM dbo.Language WHERE name = lang;
+  IF (langid IS NULL) THEN
+    RAISE '% language does not exist', lang;
+  END IF;
+
+  SELECT
+    c.id INTO categoryid
+  FROM
+    dbo.category c JOIN
+    dbo.categorylanguage cl ON c.id = cl.category
+  WHERE
+    name = categ AND language = langid;
+  IF (categoryid IS NULL) THEN
+    RAISE '"%" is not a valid category for the language %', categ, lang;
+  END IF;
+
+  -- Insert or update Programme
+  -- coalesce: if any of the supplied parameters is null, use the previous value of the column instead
+  UPDATE
+    dbo.course
+  SET
+    name     = COALESCE(courseName, name),
+    acronym  = COALESCE(courseAcr, acronym)
+  WHERE
+    name = courseName OR acronym = courseAcr;
+
+  INSERT INTO
+    dbo.Course(name, acronym)
+  SELECT
+    courseName, courseAcr
+  WHERE
+    NOT EXISTS(
+      SELECT id FROM dbo.course
+      WHERE name = courseName OR acronym = courseAcr
+    );
+
+  SELECT id INTO cid FROM dbo.Course WHERE name = courseName OR acronym = courseAcr;
+
+  -- Creating the Class
+  PERFORM
+    dbo.f_classCalendarCreate (calTerm, cid)
+  WHERE
+    NOT EXISTS(
+      SELECT id FROM dbo.Class
+      WHERE courseId = cid AND calendarTerm = calTerm
+    );
+    
+  SELECT id INTO clid FROM dbo.Class WHERE courseId = cid AND calendarTerm = calTerm;
+
+  -- Creating the Class
+  PERFORM
+    dbo.f_classSectionCalendarCreate (clid, calendarSection)
+  WHERE
+    NOT EXISTS(
+      SELECT id FROM dbo.ClassSection C
+      WHERE C.classid = clid AND C.id = calendarSection
+    );
+    
+  -- Store ClassSection's ID and Calendar
+  SELECT
+    calendar INTO cscal
+  FROM
+    dbo.ClassSection C
+  WHERE
+    C.classid = clid AND C.id = calendarSection;
+  
+  -- Deletes all of the Class Section's events related to this operation
+  DELETE FROM
+    dbo.CalendarComponent
+  WHERE id IN (
+    SELECT
+      ccs.comp_id
+    FROM
+      dbo.category c JOIN
+      dbo.categories cs ON c.id=cs.value JOIN
+      dbo.calendarcomponents ccs ON ccs.comp_id=cs.comp_id
+    WHERE
+      c.id = categoryid AND ccs.calendar_id = cscal
+  ) AND type = 'E';
+
+  RAISE NOTICE '% course, % class, % section, % category, % language, % calendar', cid, clid, calendarSection, categoryid, langid, cscal;
+END
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE PROCEDURE dbo.sp_createClassSectionEvent(
+  courseName VARCHAR(100),
+  courseAcr VARCHAR(50),
+  calendarSection VARCHAR(50),
+  calTerm VARCHAR(50),
+	summary VARCHAR(50),
+	description VARCHAR(200),
+  lang VARCHAR(10),
+	categ VARCHAR(100),
```

Comment:
An event can have multiple categories. Shouldn't this be a `VARCHAR(100)[]`?

Reply:
The `insertClassSectionEvents` operation limits this option by restricting it to only one category. The suggested operation by the Integration Group was actually a `insertClassSectionSchedule` operation which would always create `Lecture` events.

TLDR: we could apply this change, but for now I'd like to keep it simple and maybe patch it later.